app/src/main/java/com/babylon/wallet/android/data/repository/tokenprice/FiatPriceRepository.kt

```diff
@@ -148,17 +148,13 @@ class MainnetFiatPriceRepository @Inject constructor(
     }
 }
 
+// Keeps an in-memory cache of prices, so that prices of test tokens don't fluctuate between screens
+@Singleton
```

Comment:
From what I can see (at least in the case of repositories) the scope is defined when binding the dependency in `DataModule`, I'd suggest doing the same in this case.